Assets/Project/Modules/Enemies/Slime/Scripts/SlimeRefactor/SlimeAnimation.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SlimeAnimation : ISlimeComponent
```

Comment:
Don't name this class "Slime...", instead name it "SquashStretchEnemyAnimator". It no longer will inherit from a ISlimeComponent.
Instead you should make an Interface "IEnemyAnimator" with functions definition such as "PlayTakeDamage()", "PlayDeath()", "PlayMove()", etc. Then, your "SquashStretchEnemyAnimator" implementation performs the squashing and strethching.

Reply:
This way the mediator has a reference to the interface and doesn't actually know it is using a "placeholder" implementation. In the future we will make other "...EnemyAnimator" classes to handle better animations instead of just tweens